lib/excon/ssl_socket.rb

```diff
@@ -116,20 +116,17 @@ def initialize(data = {})
 
       begin
         if @nonblock
-          loop do
-            begin
-              @socket.connect_nonblock
-              break # connect succeeded
-            rescue OpenSSL::SSL::SSLError => error
-              # would block, rescue and retry as select is non-helpful
-              raise error unless error.message == 'read would block'
-            end
+          begin
+            @socket.connect_nonblock
+          rescue IO::WaitReadable
+            IO.select([@socket])
```

Comment:
You can ignore my comment on the 3rd argument to IO.select.  Even thought it's called **exceptfs**, it's for exceptional conditions not exceptions/errors as I've always assumed.  This man page has more but it's basically only used for out-of-band data which doesn't apply in this case: http://man7.org/linux/man-pages/man2/select_tut.2.html

Given that, your patch looks good to me.